test/Microsoft.DotNet.ProjectModel.Tests/LockFileReaderTests.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+using FluentAssertions;
+using Microsoft.DotNet.ProjectModel.Graph;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.DotNet.ProjectModel.Tests
+{
+    public class LockFileReaderTests : TestBase
+    {
+        [Fact]
+        public void ReadLockFileReadsLibrariesWithPathProperty()
+        {
+            // Arrange
+            var lockFileJson = @"
+            {
+              ""libraries"": {
+                ""PackageA/1.0.1-Alpha"": {
+                  ""sha512"": ""FAKE-HASH"",
+                  ""type"": ""package"",
+                  ""path"": ""PackageA/1.0.1-beta-PATH""
+                },
+                ""ProjectA/1.0.2-Beta"": {
+                  ""type"": ""project"",
+                  ""path"": ""ProjectA-PATH""
+                }
+              }
+            }";
+
+            var root = Temp.CreateDirectory();
+            var file = root.CreateFile("project.lock.json");
+            file.WriteAllText(lockFileJson);
+
+            // Act
+            var lockFile = LockFileReader.Read(file.Path, designTime: false);
```

Comment:
Why not add an overload of Read that takes a stream and skip creating a temporary directory and a file? Even a overload that takes the string directly, if you want.